wikidatarefisland/data_access/wdqs_reader.py

```diff
@@ -28,6 +31,31 @@ def get_schemaorg_mapping(self):
 }"""
         return self.get_sparql_data(query)
 
+    def get_formatter(self, pid):
+        if self._formatter_urls == {}:
+            self._populate_formatter_urls()
+        return self._formatter_urls.get(pid)
+
+    def _populate_formatter_urls(self):
+        query = """SELECT ?property ?formatter
+WHERE {
+  ?property wdt:P1630 ?formatter
+}"""
+        self._formatter_urls = defaultdict(list)
+        for case in self.get_sparql_data(query):
+            pid = case['property']['value'].replace('http://www.wikidata.org/entity/', '')
+            self._formatter_urls[pid].append(case['formatter']['value'])
```

Comment:
What is the advantage of saving the formatter URLs to a member in this class? I might be missing something here...

Reply:
Actually I moved it to a dedicated class to have the SS4 instead of making this class super super big. Would that help?